lib/Parse/ParseExpr.cpp

```diff
@@ -538,6 +539,8 @@ ParserResult<Expr> Parser::parseExprKeyPath() {
   }
 
   if (startsWithSymbol(Tok, '.')) {
+    llvm::SaveAndRestore<Expr*> S(SwiftKeyPathRoot, rootResult.getPtrOrNull());
```

Comment:
Ah, sorry I misread this.  For some reason I didn't notice they were in completely different functions ,etc.